### DISCUSSION TITLE
Scroll to top on page view

### DISCUSSION THREAD

**Discussion by thescooble** [swup/swup#351] [Help] [answered]
I have a fixed header and when I click on a link that has swup enabled, the animation works fine, but it doesnt scroll to top of the screen for the page I linked to.
I tried doing the "known issue fix" by adding a div element "<div id="need-to-scroll-here" class="anchor"></div>", adding the CSS that offset the element by -200px as well as appended the div id (#need-to-scroll-here) to the referring URL, but is still doesn't scroll to the top.
How do I get it to scroll to the top?

**Comment by coriidor5** [comment]
I couldn't find a solution using the anchors, so I had to create a solution that only really works for 'back to top' buttons. It involves using JavaScript to manually set the scroll position. One snag I encounter is the JavaScript reloading is odd for me, so I use one JS file for my small site. If the addEventListener looks for the element and it isn't there, it could disrupt the JavaScript. I wrapped it in a try-catch block so it fails gracefully. Here is the code I used.

    function backToTop() {
        document.body.scrollTop = 0;
        document.documentElement.scrollTop = 0;
    }

    try {
        const topBtn = document.querySelector('.back-to-top');
     topBtn.addEventListener('click', backToTop);
    } catch (error) {
    }

**Comment by daun** [comment, marked as answer]
**Update:** [swup 4](https://swup.js.org/announcements/swup-4/) now comes with scroll support out of the box.

The below applies to swup 3:

You could either use the [swup scroll plugin](https://github.com/swup/scroll-plugin) that will scroll back to the top after each transition, or you could implement a very simple non-animated version of it yourself:

```js
swup.on('contentReplaced', () => window.scrollTo(0, 0))
```

Keep in mind this won't handle special cases like browser history visits or links with #hashes.

**Reply by THRY** [reply]
Is it possible to disable animated scrolling completly? i think its strange in which order the transition is working.

Swup fades the page out, then fades the new page in and only after then scrolls to top of the page. 

For me it would make sense that the page scrolls up to top before the new content fades in. Is this possible? I did not find anything in the docs or the discussions.

**Reply by daun** [reply]
@THRY The official [Scroll Plugin](https://github.com/swup/scroll-plugin) allows customizing the order: it can be configured to start scrolling up to the top while the new page is loading/transitioning.
